### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by tinygrox**
made some changes for Localization.
Is this the branch?

### CODE REVIEW

**Comment by Sigma88** [comment]
A couple of thoughts

1- I don't know how I feel about including some languages into the localization but not others, I think maybe the other languages could go into a language pack rarther than kopernicus itself

2- I think that using Localize.Format comes with a performance hit, so I guess we should format the value once and then keep using the formatted string

**Comment by tinygrox** [comment]
@Sigma88 I can only do so. And I don't quite understand what you mean by "language pack".

**Comment by Sigma88** [comment]
thank you for making the changes

>I don't quite understand what you mean by "language pack".

I mean a separate download

hardcoding only one extra language would feel weird to me.

I am not necessarily against it, but I would rather defer to @StollD  on that one

File: src/Kopernicus/Components/KopernicusSolarPanels.cs [hunk 1]
```diff
@@ -37,7 +38,16 @@ namespace Kopernicus.Components
     /// </summary>
     public class KopernicusSolarPanels : PartModule
     {
-        [KSPField(guiActive = true, guiActiveEditor = false, guiName = "Tracking Body", isPersistant = true)]
+        //Strings for Localization
+        string SP_status_DirectSunlight = Localizer.Format("#Kopernicus_UI_DirectSunlight");//"Direct Sunlight"
```
Comment (Sigma88) [inline src/Kopernicus/Components/KopernicusSolarPanels.cs]: I think the new unity might have some issues with this way of defining the variables.

you should change them into static strings

I am not sure if changing them to static is enough to fix all issues or if we should also define them at a different time

**Comment by StollD** [comment]
> I mean a separate download
> 
> hardcoding only one extra language would feel weird to me.

We *have* to ship at least one language or make that language pack a requirement. The language pack cannot change the texts that Kopernicus uses.

And I would imagine that more languages will follow as people contribute them, just that @tinygrox was only able to provide an english version to get started.

**Comment by Sigma88** [comment]
>We have to ship at least one language or make that language pack a requirement. The language pack cannot change the texts that Kopernicus uses.

What I mean is: English will be included as the default, having just one extra language would feel weird to me

**Comment by Sigma88** [comment]
hardcoding all the languages would require a new release everytime a change in the localization is required (grammar checks, new languages added, etc)

**Comment by StollD** [comment]
Ah wait, there is a chinese version too. Now that makes more sense, but I don't think it is bad or weird.

If there is another language, or a fix to one of them, you can still download a random other file, put it into GameData and Kopernicus will use it. And we don't have to release a new version immediately, we can still wait until there are enough changes that warrant a release. We provide the feature, we decide how to handle it.

File: src/Kopernicus/Components/KopernicusSolarPanels.cs (hunk 1 above)
Comment (StollD) [reply src/Kopernicus/Components/KopernicusSolarPanels.cs]: Argh, you beat me to it. No, Unity won't have issues (using the constructor in a MonoBehaviour is totally fine as long as you don't create it with `new MB();`), but it is unneccessary to store a copy of these strings for every instance of the solar panel module.

Please make them `private static`.

Comment (Sigma88) [reply src/Kopernicus/Components/KopernicusSolarPanels.cs]: @tinygrox just do what @stolld said and we can merge your PR

**Comment by Sigma88** [comment]
I would also suggest to move the localization into:

`Kopernicus/Localization/english.cfg`
and
`Kopernicus/Localization/chinese.cfg`

**Comment by tinygrox** [comment]
Is that right?

**Comment by Sigma88** [comment]
Yes, thank you!

**Comment by Sigma88** [comment]
Wrong button

**Pull request merged**